In [2]:
import pandas as pd

In [3]:
df = pd.read_csv("/Users/patrickburke/Library/CloudStorage/OneDrive-EmoryUniversity/ECON496RW/cleaned_small/factpcradditionalresponsemode.csv")

In [4]:
df.head()

PcrKey  eResponse_24
0   1472192       7701003
1  13273412       2224019
2  14548133       2224015
3  17184298       2224015
4  21699657       2224019

In [5]:
len(df.index)

60716873

In [6]:
# check for duplicates
dup_check = df.duplicated(subset="PcrKey",keep='first')
dup_check

0           False
1           False
2           False
3           False
4           False
            ...  
60716868    False
60716869    False
60716870    False
60716871    False
60716872    False
Length: 60716873, dtype: bool

In [7]:
print(dup_check.value_counts()[True])

11733883


In [8]:
dup_check = df.duplicated(subset="PcrKey",keep=False)
dup_df = df[dup_check]

In [9]:
dup_df.head()

PcrKey  eResponse_24
20  118150307       2224003
21  118150307       2224009
22  118150307       2224013
23  118150307       2224005
24  118150307       2224015

In [10]:
dup_df.loc[dup_df['PcrKey'] == 118150307]

PcrKey  eResponse_24
20  118150307       2224003
21  118150307       2224009
22  118150307       2224013
23  118150307       2224005
24  118150307       2224015

- 2224001 - Intersection Navigation-Against Normal Light Patterns
- 2224003 - Intersection Navigation-With Automated Light Changing Technology 
- 2224005 - Intersection Navigation-With Normal Light Patterns
- 2224007 - Scheduled
- 2224009 - Speed-Enhanced per Local Policy
- 2224011 - Speed-Normal Traffic
- 2224013 - Unscheduled
- 2224015 - Lights and Sirens
- 2224017 - Lights and No Sirens
- 2224019 - No Lights or Sirens
- 2224021 - Initial No Lights or Sirens, Upgraded to Lights and Sirens
- 2224023 - Initial Lights and Sirens, Downgraded to No Lights or Sirens

- Intersection Navigation-With Automated Light Changing Technology, Speed-Enhanced per Local Policy, Unscheduled, Intersection Navigation-With Normal Light Patterns, Lights and Sirens
- Only conflict here is multiple types of intersection navigation but thats actually very possible, duplicates OK

In [11]:
df.head()

PcrKey  eResponse_24
0   1472192       7701003
1  13273412       2224019
2  14548133       2224015
3  17184298       2224015
4  21699657       2224019

In [12]:
len(df.index)

60716873